video/out/vo_dmabuf_wayland.c

```diff
@@ -77,10 +77,17 @@ static bool vaapi_dmabuf_importer(struct mp_image *src, struct wlbuf_pool_entry*
     bool dmabuf_imported = false;
     /* composed has single layer */
     int layer_no = 0;
-    VAStatus status = vaExportSurfaceHandle(p->display, entry->key, VA_SURFACE_ATTRIB_MEM_TYPE_DRM_PRIME_2,
+    VAStatus status;
+
+    desc.num_objects = 0;
+    status = vaExportSurfaceHandle(p->display, entry->key, VA_SURFACE_ATTRIB_MEM_TYPE_DRM_PRIME_2,
                                             VA_EXPORT_SURFACE_COMPOSED_LAYERS | VA_EXPORT_SURFACE_READ_ONLY, &desc);
     if (status == VA_STATUS_ERROR_INVALID_SURFACE) {
         MP_VERBOSE(entry->vo, "VA export to composed layers not supported.\n");
+    } else if (status == VA_STATUS_ERROR_INVALID_DISPLAY) {
+        MP_VERBOSE(entry->vo, "VA display is invalid.\n");
+    } else if (status == VA_STATUS_ERROR_UNSUPPORTED_MEMORY_TYPE) {
+        MP_VERBOSE(entry->vo, "Driver does not support drm prime memory type.\n");
     } else if (!vo_wayland_supported_format(entry->vo, desc.layers[0].drm_format, desc.objects[0].drm_format_modifier)) {
```

Comment:
the access to data inside `desc` here happens before you exhaustively checked the return code (`CHECK_VA_STATUS` below).
I think you'll need to split these apart